src/error/specification

```diff
@@ -17,1331 +17,34 @@ ignore:true
 :It is for errors which are not relevant anymore
 :an empty line will start the next error specification
 
-number:1
-ingroup:modules
-description:could not load module, dlopen failed
-module:dl
-severity:warning
-class:FailedLoadModuleError
-extends:ModuleError
+number:01100
+description:Resource
+macro:RESOURCE_CODE
```

Comment:
The previous system was better. Here should be just one word, and then some common prefix should make that the macros do not clash with other macros.

Reply:
Do you have an suggestion for the prefix? Warning/Error should not be in the prefix

Reply:
Why should error not be in the prefix? It is an error code after all...

Reply:
Because it can also be used as a warning

Reply:
But as you said yourself below, errors and warnings should be interchangeable. So essentially a warning is just a special kind of error. If you really want to keep the semantics neat, just generate to macros one with the prefix `ELEKTRA_ERROR_` and one with `ELEKTRA_WARNING_`.

Reply:
Alternatively we could encode the error directly in the issuing macro, e.g. `ELEKTRA_ADD_RESOURCE_WARNING` or `ELEKTRA_SET_RESOURCE_ERROR`. This macro would simply have one parameter less.

As there are so few errors/warnings now, which all can be described with one/two words, this should work?

I am not sure though if this will conflict with the improvements of a smaller kdberrors.h as suggested by @kodebach.

Reply:
Just because the number of errors are so few at the moment, it does not mean that this will not grow again. I also do not think that this is more intuitive for devs. The parameter omission comes with the price for knowing the exact macro name.

Reply:
I hope they will not grow again!

Reply:
> The parameter omission comes with the price for knowing the exact macro name.

Does it really matter if you need to know the error macro code name or error macro name?

The main difference is that not two but only one macro need to be prefixed, leading to an overall shorter call:

```
ELEKTRA_SET_RESOURCE_ERROR(...
vs.
ELEKTRA_SET_ERROR(ELEKTRA_RESOURCE_CODE, ...
```

The last time we discussed it, however, there were people against this approach. One disadvantage is, that the code cannot be calculated. This should be very minor, though.

Please do not forget to write design decisions about these topics, so that we have it documented. Please update #2572.

Reply:
> One disadvantage is, that the code cannot be calculated

The current code also doesn't allow calculated codes. In fact they have to be known to the preprocessor not just to the compiler. If codes aren't statically known, you have to use the `elektraTriggerError` function, but that means you don't get compile time error for unspecified error codes anymore.